### DISCUSSION TITLE
Using ANTLR4 on pyscript

### DISCUSSION THREAD

**Discussion by jl-mobitech** [pyscript/pyscript#1686] [Q&A]
I am developing a calculator application that uses MathLive (https://cortexjs.io/mathlive/) as its front-end. Now, on MathLive, I can input any math equations and it would be automatically displayed as latex. Now to process this, I would use sympy. Browsing sympy's documentation, I read that I can parse latex but it needs ANTLR4 (https://docs.sympy.org/latest/modules/parsing.html#experimental-mathrm-latex-parsing). Here's my code to include sympy and antlr4:

```html
 <py-config>
        packages = ['sympy', 'antlr4-python3-runtime']
 </py-config>

<py-script src="/actual_script.py"></py-script>
```
  
Now when I try to parse a latex using this code (this code is some of the contents of actual_script.py):
```python
import sympy as sp
from sympy.parsing.latex import parse_latex

latex_string = r"\Delta(k) = \frac{\rho_1-\rho_2}{\rho_1 + \rho_2} gk + \frac{\gamma k^3}{\rho_1 + \rho_2} - \frac{\rho_1 \rho_2}{(\rho_1 + \rho_2)^2} U^2 k^2"
equation = parse_latex(latex_string)
print(equation)
```

It throws this errror:

![image](https://github.com/pyscript/pyscript/assets/93818433/7c7db4cc-242a-4ec0-b758-70836d7ae1d4)

I'm wondering, is my requirements possible using pyscript? Any ideas how I can proceed?

Another note: I also tried installing using .whl file but the same error occurs. ANTLR4 is also in PyPI: https://pypi.org/project/antlr4-python3-runtime/

**Comment by Neon22** [comment]
There is a pure python implementation here: https://pypi.org/project/antlr4-grun/
I haven't been able to confirm if this would work or if it has the functionality you need.

**Comment by JeffersGlass** [comment]
I looks like `antr4-python3-rutime` just provides an interface to the underlying executable (writtenin Java), as well as an installer that installs Java and the executable if necessary.
